### PR TITLE
Wrap detection module

### PR DESCRIPTION
**Pull request by AKS1996**
For now, will tell whether april tag is present in the image or not. Will add other features soon. [Refer here](https://github.com/AKS1996/GSOC-Prep/tree/simple_graphics/visp-java-demos/visp-detection-module-demo) for an Eclipse demo tutorial.

### CODE REVIEW

**Comment by AKS1996** [comment]
Depends on #395 to work

**Comment by AKS1996** [comment]
@fspindle Can you see why is the build failing for `gcc`? I'm unable to reproduce the error on my desktop.

**Comment by AKS1996** [comment]
@fspindle See the java equivalent of this [apriltag detection demo](http://visp-doc.inria.fr/doxygen/visp-daily/tutorial-detection-apriltag.html) here in this [Eclipse project](https://github.com/AKS1996/GSOC-Prep/tree/simple_graphics/visp-java-demos/visp-detection-module-demo). 

screenshot from 2018-07-10 12-15-05

**Comment by AKS1996** [comment]
@fspindle Are you able to run Eclipse project?

**Comment by fspindle** [comment]
Good work.
I think that there is a small bug in `VpDisplay::displayFrame()` visible mainly on the blue arrow corresponding to the z-axis. To reproduce just use `visp/tutorial/detection/tag/AprilTag.pgm` image.
Using this `AprilTag.pgm` image I have also an extra blue line starting from the left-top corner (0,0).

**Comment by AKS1996** [comment]
Ok. I'll look into it

**Comment by fspindle** [comment]
Yes, very strange that `chocolatey` fails to install `ant`.

**Comment by AKS1996** [comment]
Minor JSON typo :sweat_smile:

**Comment by AKS1996** [comment]
Depends on #411 for travis build

**Comment by AKS1996** [comment]
Its complete now. Ready for merging

File: modules/java/generator/src/cpp/listconverters.cpp [hunk 1]
```diff
@@ -26,10 +42,23 @@ std::vector<float> List_to_vector_float(JNIEnv* env, jfloatArray arr)
     V.push_back(body[i]);
 
   env->ReleaseFloatArrayElements(arr, body, 0);
-  
   return V;
 }
 
+jlongArray vector_vpHomogeneousMatrix_to_List(JNIEnv* env, std::vector<vpHomogeneousMatrix> V)
```
Comment (s-trinh) [inline modules/java/generator/src/cpp/listconverters.cpp]: Is this correct? `jlongArray` should hold signed 64 bits integer. Is it not `jdoubleArray` instead?

Comment (s-trinh) [inline modules/java/generator/src/cpp/listconverters.cpp]: I think you can use `std::vector<vpHomogeneousMatrix>& V` to avoid a copy.

**Pull request merged**